src/rules/no-duplicates.js

```diff
@@ -2,21 +2,198 @@ import resolve from 'eslint-module-utils/resolve'
 import docsUrl from '../docsUrl'
 
 function checkImports(imported, context) {
-  for (let [module, nodes] of imported.entries()) {
-    if (nodes.size > 1) {
-      for (let node of nodes) {
-        context.report(node, `'${module}' imported multiple times.`)
+  for (const [module, nodes] of imported.entries()) {
+    if (nodes.length > 1) {
+      const message = `'${module}' imported multiple times.`
+      const [first, ...rest] = nodes
+      const sourceCode = context.getSourceCode()
+      const fix = getFix(first, rest, sourceCode)
+
+      context.report({
+        node: first.source,
+        message,
+        fix, // Attach the autofix (if any) to the first import.
+      })
+
+      for (const node of rest) {
+        context.report({
+          node: node.source,
+          message,
+        })
+      }
+    }
+  }
+}
+
+function getFix(first, rest, sourceCode) {
+  // Sorry ESLint <= 3 users, no autofix for you.
+  if (typeof sourceCode.getCommentsBefore !== 'function') {
+    return undefined
+  }
+
+  const defaultImportNames = new Set(
+    [first, ...rest].map(getDefaultImportName).filter(Boolean)
+  )
+
+  // Bail if there are multiple different default import names – it's up to the
+  // user to choose which one to keep.
+  if (defaultImportNames.size > 1) {
+    return undefined
+  }
+
+  // It's not obvious what the user wants to do with comments associated with
+  // duplicate imports, so skip imports with comments when autofixing.
+  const restWithoutComments = rest.filter(node => !(
+      hasCommentBefore(node, sourceCode) ||
+      hasCommentAfter(node, sourceCode) ||
+      hasCommentInsideNonSpecifiers(node, sourceCode)
+  ))
+
+  const specifiers = restWithoutComments
+    .map(node => {
+      const tokens = sourceCode.getTokens(node)
+      const openBrace = tokens.find(token => isPunctuator(token, '{'))
+      const closeBrace = tokens.find(token => isPunctuator(token, '}'))
+
+      if (openBrace == null || closeBrace == null) {
+        return undefined
+      }
+
+      return {
+        importNode: node,
+        text: sourceCode.text.slice(openBrace.range[1], closeBrace.range[0]),
+        hasTrailingComma: isPunctuator(sourceCode.getTokenBefore(closeBrace), ','),
+        isEmpty: !hasSpecifiers(node),
+      }
+    })
+    .filter(Boolean)
+
+  const unnecessaryImports = restWithoutComments.filter(node =>
+    !hasSpecifiers(node) &&
+    !specifiers.some(specifier => specifier.importNode === node)
+  )
+
+  const shouldAddDefault = getDefaultImportName(first) == null && defaultImportNames.size === 1
+  const shouldAddSpecifiers = specifiers.length > 0
+  const shouldRemoveUnnecessary = unnecessaryImports.length > 0
+
+  if (!(shouldAddDefault || shouldAddSpecifiers || shouldRemoveUnnecessary)) {
+    return undefined
+  }
+
+  return function* (fixer) {
```

Comment:
does this really need to be a generator?

Reply:
I think it's possible to make an array, `const fixes = []`, and `.push()` to it instead of `yield` and return that at the end. I can do that if you think that's better (and it works).

Reply:
Confirmed – the array version works, too. Personally I like the generator here, but let me know which way you want it.

Reply:
I'd prefer the array version, I find it easier to understand.

Reply:
Pushed the array version!